### PR TITLE
MODINV-162, MODINV-160: Handle HRID for instance and item.

### PR DESCRIPTION
**Pull request by bohdan-suprun**
https://issues.folio.org/browse/MODINV-160 - HRID handling in Inventory for Instance records.
https://issues.folio.org/browse/MODINV-162 - HRID handling in Inventory for Item records.

### Changes:
* Added Item/Instance fake pre-processors to generate a HRID if nothing has specified;
* Disallow changing of hrid for item and instance;
* Added unit tests.

### CODE REVIEW

File: src/test/java/api/InstancesApiExamples.java [hunk 1]
```diff
@@ -150,20 +151,24 @@ public void canCreateAnInstance()
     dublinCoreContextLinkRespectsWayResourceWasReached(createdInstance);
     selfLinkRespectsWayResourceWasReached(createdInstance);
     selfLinkShouldBeReachable(createdInstance);
+
+    assertThat(createdInstance.getString("hrid"), notNullValue());
```
Comment (marcjohnson-kint) [inline src/test/java/api/InstancesApiExamples.java]: I think this test should be renamed to `canCreateInstanceWithoutAnIDAndHRID`

File: src/test/java/api/InstancesApiExamples.java [hunk 2]
```diff
@@ -150,20 +151,24 @@ public void canCreateAnInstance()
...
   }
 
   @Test
-  public void canCreateAnInstanceWithAnID()
+  public void canCreateAnInstanceWithAnIDAndHrid()
```
Comment (marcjohnson-kint) [inline src/test/java/api/InstancesApiExamples.java]:
```suggestion
  public void canCreateAnInstanceWithAnIDAndHRID()
```

File: src/test/java/api/InstancesApiExamples.java [hunk 3]
```diff
@@ -826,6 +832,31 @@ public void cannotFindAnUnknownInstance()
     assertThat(getResponse.getStatusCode(), is(404));
   }
 
+  @Test
+  public void cannotUpdateHrid() throws Exception {
```
Comment (marcjohnson-kint) [inline src/test/java/api/InstancesApiExamples.java]: What happens if the HRID is not specified in the update?

Comment (marcjohnson-kint) [inline src/test/java/api/InstancesApiExamples.java]:
```suggestion
  public void cannotUpdateHRID() throws Exception {
```

File: src/test/java/api/items/ItemApiExamples.java [hunk 4]
```diff
@@ -144,16 +145,18 @@ public void canCreateAnItem()
 
     selfLinkRespectsWayResourceWasReached(createdItem);
     selfLinkShouldBeReachable(createdItem);
+    assertThat(createdItem.getString("hrid"), notNullValue());
   }
 
   @Test
-  public void canCreateItemWithAnID()
+  public void canCreateItemWithAnIDAndHrid()
```
Comment (marcjohnson-kint) [inline src/test/java/api/items/ItemApiExamples.java]:
```suggestion
  public void canCreateItemWithAnIDAndHRID()
```

File: src/test/java/api/items/ItemApiExamples.java [hunk 5]
```diff
@@ -1311,6 +1317,40 @@ public void itemHasNoLastCheckInPropertiesWhenNotSet() throws Exception {
     assertThat(actualItem.getJsonObject(LAST_CHECK_IN_FIELD), is(nullValue()));
   }
 
+  @Test
+  public void cannotUpdateHrid() throws Exception {
```
Comment (marcjohnson-kint) [inline src/test/java/api/items/ItemApiExamples.java]:
```suggestion
  public void cannotChangeHRID() throws Exception {
```

File: src/test/java/api/InstancesApiExamples.java (hunk 3 above)
Comment (bohdan-suprun) [reply src/test/java/api/InstancesApiExamples.java]: The update will be rejected.

Comment (marcjohnson-kint) [reply src/test/java/api/InstancesApiExamples.java]: Please can a test be added to demonstrate that a HRID cannot be removed (for both instance and item).

File: src/test/java/api/items/ItemApiExamples.java [hunk 6]
```diff
@@ -78,7 +79,7 @@ public ItemApiExamples() throws MalformedURLException {
   }
 
   @Test
-  public void canCreateAnItem()
+  public void canCreateAnItemWithoutIdAndHRID()
```
Comment (marcjohnson-kint) [inline src/test/java/api/items/ItemApiExamples.java]:
```suggestion
  public void canCreateAnItemWithoutIDAndHRID()
```

**Comment by marcjohnson-kint** [comment]
Does the instance batch API allow for instances to be updated?

**Comment by bohdan-suprun** [comment]
> Does the instance batch API allow for instances to be updated?

It seems not, there is only create batch method. It executes the storage batch method + saves instance relationships.

File: src/test/java/api/items/ItemApiExamples.java (hunk 6 above)
Comment (marcjohnson-kint) [reply src/test/java/api/items/ItemApiExamples.java]: The `Id` needs capitalising to `ID`

**Pull request merged**